traffic_ops/app/lib/API/User.pm

```diff
@@ -174,7 +179,9 @@ sub update {
 		registration_sent 		=> ( $params->{registrationSent} ) ? 1 : 0,
 		role 					=> $params->{role},
 		state_or_province 		=> $params->{stateOrProvince},
-		username 				=> $params->{username}
+		username 				=> $params->{username},
+		tenant_id 				=> $tenant_id
```

Comment:
I think you can just do this to get the desired result:

tenant_id  => $params->{tenantId}

and if it's not provided, it will set it to  undef, otherwise it will set it accordingly